### PR TITLE
Integrate WindowsAI feature branch with DML EP features and DML 1.9

### PR DESCRIPTION
**Pull request by fdwr**
**Description**:
- Integrates DML EP features from [WindowsAI feature branch](https://github.com/microsoft/onnxruntime/tree/WindowsAI):
  - **New operators**: Trilu-14, HardSwish-15, Castlike-15, Gelu.com.microsoft-1
  - **Updated operators**: SoftMax-13, LogSoftMax-13, HardMax-13, QuantizeLinear-13, DequantizeLinear-13, Resize-13, Split-13, Squeeze-13, Unsqueeze-13, ReduceSum-13, CumSum-13, Add-14, Sub-14, Mul-14, Div-14, Reshape-14, Relu-14, BatchNormalization-15, {GRU-14, LSTM-14, RNN-14} with fallback to CPU EP when layout=1, BatchNormalization-14 with fallblack to CPU EP when training_mode=1.
  - **Verified but unsupported**: Optional-15 via CPU kernel
- Updates build files to enable external DML DLL build (point to specific one):
  - cmake/external/dml.cmake
  - cmake/onnxruntime_providers.cmake
  - tools/ci_build/build.py
- Updates DirectML.dll to [1.9.0 preview](https://nuget.org/packages/Microsoft.AI.DirectML.Preview).

### CODE REVIEW

**Comment by fdwr** [comment]
@snnn : There are a few changes to build files by @jstoecker in this feature RI:

- cmake/external/dml.cmake
- cmake/onnxruntime_providers.cmake
- tools/ci_build/build.py

**Comment by adtsai** [comment]
> {

Should GELU not also be part of this union?

---
Refers to: onnxruntime/core/providers/dml/DmlExecutionProvider/src/External/DirectMLHelpers/ApiHelpers.h:7 in 04dd663. [](commit_id = 04dd6639de1d2dcb8ff410994ebc68000ab448a9, deletion_comment = False)

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/External/DirectMLHelpers/ApiTraits.h [hunk 1]
```diff
@@ -2532,6 +2630,9 @@ inline gsl::czstring ToString(DML_OPERATOR_TYPE value)
     case DML_OPERATOR_ELEMENT_WISE_QUANTIZED_LINEAR_ADD: return "DML_OPERATOR_ELEMENT_WISE_QUANTIZED_LINEAR_ADD";
     case DML_OPERATOR_ROI_ALIGN_GRAD: return "DML_OPERATOR_ROI_ALIGN_GRAD";
     case DML_OPERATOR_BATCH_NORMALIZATION_TRAINING: return "DML_OPERATOR_BATCH_NORMALIZATION_TRAINING";
+    case DML_OPERATOR_RESAMPLE2: return "DML_OPERATOR_RESAMPLE2";
+    case DML_OPERATOR_RESAMPLE_GRAD1: return "DML_OPERATOR_RESAMPLE_GRAD1";
```
Comment (adtsai) [inline onnxruntime/core/providers/dml/DmlExecutionProvider/src/External/DirectMLHelpers/ApiTraits.h:2634]: Missing the new activation functions? (Has this file diverged from what we have in WAI?)

**Review by adtsai — APPROVED** [review]

File: onnxruntime/core/providers/dml/DmlExecutionProvider/src/External/DirectMLHelpers/ApiTraits.h (hunk 1 above)
Comment (fdwr) [reply onnxruntime/core/providers/dml/DmlExecutionProvider/src/External/DirectMLHelpers/ApiTraits.h:2634]: @sumitsays: Do you recall adding Gelu here or not?

Comment (fdwr) [reply onnxruntime/core/providers/dml/DmlExecutionProvider/src/External/DirectMLHelpers/ApiTraits.h:2634]: @adtsai Actually it's evidently a bug in `dml\Tools\GenerateSchema.ps1`, as it's not present in the DML side either. For some reason in DirectMLSchema.json, all our operators are under "operators:", *except* the activation operators which are under "activationOperators:", except for `ACTIVATION_RELU_GRAD` which is actually under "operators" 🙃. I don't know why activationOperators are deemed any more special than elementwise or reduction to deserve their entire own category, but I feel they should all just be put under `operators`, and we add a flag to the entries like `isActivation: true`, like we have for other flags (e.g. `"dimensionsGenericAndMatching": true` or `"private": true`...).

So I'll leave the `switch` as-is, since it's not causing any issues, and it leaves the two headers consistent (but you're right about the union missing GELU).

**Review by sumitsays — APPROVED** [review]
